sdk/eventgrid/azure-messaging-eventgrid-cloudnative-cloudevents/src/main/java/com/azure/messaging/eventgrid/cloudnative/cloudevents/EventGridCloudNativeEventPublisher.java

```diff
@@ -0,0 +1,201 @@
+     */
+    public static Mono<Void> sendEventAsync(EventGridPublisherAsyncClient<com.azure.core.models.CloudEvent> asyncClient,
+        CloudEvent event) {
+        return asyncClient.sendEvent(toEventGridCloudEvent(event));
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain.
+     *
+     * @param syncClient a service client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     *
+     * @throws NullPointerException if {@code events} is null.
+     */
+    public static void sendEvents(EventGridPublisherClient<com.azure.core.models.CloudEvent> syncClient,
+        Iterable<CloudEvent> events) {
+        syncClient.sendEvents(toEventGridCloudEvents(events));
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain.
+     *
+     * @param asyncClient a service asynchronous client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     * @return a Mono that completes when the events are sent to the service.
+     *
+     * @throws NullPointerException if {@code events} is null.
+     */
+    public static Mono<Void> sendEventsAsync(
+        EventGridPublisherAsyncClient<com.azure.core.models.CloudEvent> asyncClient, Iterable<CloudEvent> events) {
+        return asyncClient.sendEvents(toEventGridCloudEvents(events));
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain and gives the response issued by EventGrid.
+     *
+     * @param syncClient a service client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     * @param context the context to use along the pipeline.
+     * @return the response from the EventGrid service.
+     *
+     * @throws NullPointerException if {@code events} is null.
+     */
+    public static Response<Void> sendEventsWithResponse(EventGridPublisherClient<com.azure.core.models.CloudEvent> syncClient,
+        Iterable<CloudEvent> events, Context context) {
+        return syncClient.sendEventsWithResponse(toEventGridCloudEvents(events), context);
+    }
+
+    /**
+     * Publishes the given native cloud events to the set topic or domain and gives the response issued by EventGrid.
+     *
+     * @param asyncClient a service asynchronous client that publishes events to an EventGrid topic or domain.
+     * @param events the native cloud events to publish.
+     * @return the response from the EventGrid service.
+     *
+     * @throws NullPointerException if {@code events} is null.
+     */
+    public static Mono<Response<Void>> sendEventsWithResponseAsync(
+        EventGridPublisherAsyncClient<com.azure.core.models.CloudEvent> asyncClient, Iterable<CloudEvent> events) {
+        return asyncClient.sendEventsWithResponse(toEventGridCloudEvents(events));
```

Comment:
toEventGridCloudEvents can throw NPE. So, all async methods should return a Mono.error instead of throwing.